src/mixins/subscriptionComponent.js

```diff
@@ -35,8 +35,17 @@ export default {
   ],
   beforeMount () {
     this.$workflowService.subscribe(this)
+    this.$workflowService.startSubscriptions()
   },
   beforeDestroy () {
     this.$workflowService.unsubscribe(this)
+  },
+  watch: {
+    query () {
+      // if the query changes, unsubscribe & re-subscribe
+      this.$workflowService.unsubscribe(this)
+      this.$workflowService.subscribe(this)
+      this.$workflowService.startSubscriptions()
+    }
```

Comment:
Ah heck, something I hadn't thought about is what happens when there are multiple "views" in the workspace.

Currently we always initiate a workspace with a single view (because we don't remember the previous tab layout and haven't got default layouts sorted yet). But soon we will be initiating views with multiple components.

I think that calling `startSubscriptions` in either `beforeMount` or `watch.query` would result in subscribing and unsubscribing multiple times as each view registers its subscription in turn. This is probably why this call was bound to navigation events rather than component lifecycle events (as we did briefly remember tab layout by accident in an earlier version).

So we want to delay running `$workflowService.startSubscriptions()` until `$workflowService.subscribe(this)` has run for all views, not just the first. I'm not sure how to do this without introducing an arbitrary timeout.

Perhaps the parent view should be responsible for registering subscriptions before the child component is mounted?

Reply:
Is this ok to go in, with a follow-up issue for dealing with multiple subscriptions on page load?

Reply:
Aye but in the Vue 3 upgrade this also fixes the worse problem of tasks not loading when navigating between different workflows in the Workspace view

Reply:
Where did this discussion end up?

Reply:
https://github.com/cylc/cylc-ui/issues/1278